src/components/computing-status/use-computing-status.ts

```diff
@@ -76,43 +64,88 @@ function isWorthUpdate(
     return false;
 }
 
+const genetateStatusFromComputingType = (base: string) => ({
+    invalidations: [`${base}_status`, `${base}_failed`],
+    completions: [`${base}Result`, `${base}_failed`],
+});
+
+type AnalysisStatusType = {
+    invalidations: string[];
+    completions: string[];
+};
+type ComputingTypeToStatusMapperType = Record<
+    ComputingType,
+    AnalysisStatusType
+>;
+const computingTypeToStatusMapper: ComputingTypeToStatusMapperType = {
+    [ComputingType.LOAD_FLOW]: genetateStatusFromComputingType('loadflow'),
+    [ComputingType.SECURITY_ANALYSIS]:
+        genetateStatusFromComputingType('securityAnalysis'),
+    [ComputingType.SENSITIVITY_ANALYSIS]: genetateStatusFromComputingType(
+        'sensitivityAnalysis'
+    ),
+    [ComputingType.NON_EVACUATED_ENERGY_ANALYSIS]:
+        genetateStatusFromComputingType('nonEvacuatedEnergy'),
+    [ComputingType.SHORT_CIRCUIT]: genetateStatusFromComputingType(
+        'shortCircuitAnalysis'
+    ),
+    [ComputingType.SHORT_CIRCUIT_ONE_BUS]: genetateStatusFromComputingType(
+        'oneBusShortCircuitAnalysis'
+    ),
+    [ComputingType.DYNAMIC_SIMULATION]:
+        genetateStatusFromComputingType('dynamicSimulation'),
+    [ComputingType.VOLTAGE_INITIALIZATION]:
+        genetateStatusFromComputingType('voltageInit'),
+    [ComputingType.STATE_ESTIMATION]:
+        genetateStatusFromComputingType('stateEstimation'),
+};
+
+interface UseComputingStatusProps {
+    (
+        fetcher: (studyUuid: UUID, nodeUuid: UUID) => Promise<string>,
+        resultConversion: (x: string) => RunningStatus,
+        computingType: ComputingType,
+        optionalServiceAvailabilityStatus?: OptionalServicesStatus
+    ): void;
+}
 /**
  *  this hook loads <computingType> state into redux, then keeps it updated according to notifications
- * @param studyUuid current study uuid
- * @param nodeUuid current node uuid
  * @param fetcher method fetching current <computingType> state
  * @param invalidations when receiving notifications, if updateType is included in <invalidations>, this hook will update
  * @param resultConversion converts <fetcher> result to RunningStatus
  * @param computingType ComputingType targeted by this hook
  * @param optionalServiceAvailabilityStatus status of an optional service
  */
 export const useComputingStatus: UseComputingStatusProps = (
-    studyUuid,
-    nodeUuid,
     fetcher,
-    invalidations,
-    completions,
     resultConversion,
     computingType,
     optionalServiceAvailabilityStatus = OptionalServicesStatus.Up
 ) => {
+    const nodeUuid = useSelector(
+        (state: ReduxState) => state.currentTreeNode?.id
+    );
+    const studyUuid = decodeURIComponent(useParams().studyUuid ?? '') as UUID;
```

Comment:
makes me wonder why we are storing study uuid in redux store

Reply:
I would prefer using the one from the store instead of the one from the URL.
It's a matter of responsibility for me, why a custom hook for computation status should be able to parse something in the URL... ? seems strange
Otherwise it's not strange if this code access to global states no ?

Reply:
@jonenst what is your point of view on this ?

Reply:
useParams is obviously less flexible than taking the value from something else from the react context (e.g. a store, or our own contextprovider) because it means you force the code that can be deeply nested to know that there is a <Route match=..> high above and that the url has some kind of known structure.

I don't think it matters much in this case though. Maybe juste add a wrapper useStudyUuid (implemented with useParams) so we can track more easily which component access this value, and change it if we ever need to display this components with a different study than the one in the url or in a context without a router